pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -727,6 +727,26 @@ func (m *kubeGenericRuntimeManager) SyncPod(pod *v1.Pod, podStatus *kubecontaine
 		return
 	}
 
+	// Waiting all volumes mounted. if true, continue to create container; otherwise, select until timeout
+	waitVolumesAllMountedResult := kubecontainer.NewSyncResult(kubecontainer.WaitMount, podSandboxID)
+	result.AddSyncResult(waitVolumesAllMountedResult)
+	select {
+	case allMounted, open := <-volumesAllMounted:
+		if !open {
+			message := fmt.Sprintf("pod %v volumesAllMounted channel is closed. Exiting kubeGenericRuntimeManager SyncPod. : %v", format.Pod(pod), err)
+			klog.Error(message)
+			configPodSandboxResult.Fail(kubecontainer.ErrWaitMount, message)
+			return
+		}
+
+		if !allMounted {
+			message := fmt.Sprintf("pod %v can not wait all volumes mounted.", format.Pod(pod))
+			klog.Error(message)
+			configPodSandboxResult.Fail(kubecontainer.ErrWaitMount, message)
+			return
+		}
+	}
```

Comment:
If we need add some logic about time out

Reply:
I think we do not need timeout logic. Because `WaitForAttachAndMount` has haved `podAttachAndMountTimeout`, and only return error when timeout expired. Add we can write "false" to `volumeMountedChannel` if `WaitForAttachAndMount` return error. So we do not need.

Please see: 
https://github.com/kubernetes/kubernetes/blob/0c9964fac3576534a1cd6a9b7429a417972284b1/pkg/kubelet/volumemanager/volume_manager.go#L360

Reply:
> I think we do not need timeout logic. Because `WaitForAttachAndMount` has haved `podAttachAndMountTimeout`, and only return error when timeout expired. Add we can write "false" to `volumeMountedChannel` if `WaitForAttachAndMount` return error. So we do not need.
> 
> Please see:
> https://github.com/kubernetes/kubernetes/blob/0c9964fac3576534a1cd6a9b7429a417972284b1/pkg/kubelet/volumemanager/volume_manager.go#L360

So, how about using the `chan` directly? `select` seems to be useless.

Reply:
Oh, thank for your reminding. 
I add timeout logic before，ignore deleting `select` .